stability: panic in context with go1.7

`beta` cluster running `e5e7518dc3edb358ddbddb39e96d9512775540b2`

Two nodes were upgraded first and crashed within 1h.
The panic shows backtrace as the first few stack frames, but that's due to the recover call.
The actual panic ends with `context.propagateCancel`, both triggered originally by `PeriodicallyCheckForUpdates`.

We have not seen this on builds before 1.7, and the `/go/src/golang.org/x/net/context/go17.go` trace seems to indicate a switch to the built-in library code path for `Context`. This could be a red herring.

on `104.196.54.82`:
```
I160803 15:10:21.972426 cli/start.go:334  CockroachDB beta-20160728-258-ge5e7518 (linux amd64, built 2016/08/03 13:24:33, go1.7rc3)
...
[bcd] 2016/08/03 16:11:18 Tracer finished execution
[bcd] 2016/08/03 16:11:18 Trace request complete
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xbd7c86]

goroutine 111 [running]:
panic(0x1535620, 0xc420010060)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/cli.initBacktrace.func2(0x1535620, 0xc420010060)
        /go/src/github.com/cockroachdb/cockroach/cli/backtrace.go:98 +0xca
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Recover(0xc42038c1b0)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:173 +0x56
panic(0x1535620, 0xc420010060)
        /usr/local/go/src/runtime/panic.go:458 +0x243
context.propagateCancel(0x0, 0x0, 0x1fba300, 0xc442465780)
        /usr/local/go/src/context/context.go:245 +0x26
context.WithCancel(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/context/context.go:231 +0x180
golang.org/x/net/context.WithCancel(0x0, 0x0, 0x0, 0x0, 0x1fcb0c0)
        /go/src/golang.org/x/net/context/go17.go:33 +0x5d
github.com/cockroachdb/cockroach/sql.NewSession(0x0, 0x0, 0x0, 0x0, 0x1664f99, 0x4, 0xc4202a8600, 0x0, 0x0, 0xc459dc56e0)
        /go/src/github.com/cockroachdb/cockroach/sql/session.go:94 +0x318
github.com/cockroachdb/cockroach/server.(*adminServer).GetUIData(0xc42007b5b0, 0x0, 0x0, 0xc457f67da0, 0xc42892ee48, 0xc44d92df60, 0xc42017cea0)
        /go/src/github.com/cockroachdb/cockroach/server/admin.go:641 +0x86
github.com/cockroachdb/cockroach/server.(*Server).usageReportingEnabled(0xc42007b340, 0xc457f67e60)
        /go/src/github.com/cockroachdb/cockroach/server/updates.go:213 +0xce
github.com/cockroachdb/cockroach/server.(*Server).maybeReportUsage(0xc42007b340, 0x351bca41811, 0x22e5940)
        /go/src/github.com/cockroachdb/cockroach/server/updates.go:238 +0x47
github.com/cockroachdb/cockroach/server.(*Server).PeriodicallyCheckForUpdates.func1()
        /go/src/github.com/cockroachdb/cockroach/server/updates.go:97 +0xa8
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc42038c1b0, 0xc4213dfdc0)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66
```

```
I160803 15:44:54.851751 cli/start.go:334  CockroachDB beta-20160728-258-ge5e7518 (linux amd64, built 2016/08/03 13:24:33, go1.7rc3)
...
[bcd] 2016/08/03 16:45:49 Tracer failed to run: fork/exec /opt/backtrace/bin/ptrace: cannot allocate memory
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xbd7c86]

goroutine 78 [running]:
panic(0x1535620, 0xc420010060)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/cockroachdb/cockroach/cli.initBacktrace.func2(0x1535620, 0xc420010060)
        /go/src/github.com/cockroachdb/cockroach/cli/backtrace.go:98 +0xca
github.com/cockroachdb/cockroach/util/stop.(*Stopper).Recover(0xc42035f050)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:173 +0x56
panic(0x1535620, 0xc420010060)
        /usr/local/go/src/runtime/panic.go:458 +0x243
context.propagateCancel(0x0, 0x0, 0x1fba300, 0xc5cf1eca00)
        /usr/local/go/src/context/context.go:245 +0x26
context.WithCancel(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/go/src/context/context.go:231 +0x180
golang.org/x/net/context.WithCancel(0x0, 0x0, 0x0, 0x0, 0x1fcb0c0)
        /go/src/golang.org/x/net/context/go17.go:33 +0x5d
github.com/cockroachdb/cockroach/sql.NewSession(0x0, 0x0, 0x0, 0x0, 0x1664f99, 0x4, 0xc42006a700, 0x0, 0x0, 0xc57a348240)
        /go/src/github.com/cockroachdb/cockroach/sql/session.go:94 +0x318
github.com/cockroachdb/cockroach/server.(*adminServer).GetUIData(0xc420307df0, 0x0, 0x0, 0xc4d9cbfda0, 0xc4e6fb4a88, 0xc428935f50, 0xc420060b60)
        /go/src/github.com/cockroachdb/cockroach/server/admin.go:641 +0x86
github.com/cockroachdb/cockroach/server.(*Server).usageReportingEnabled(0xc420307b80, 0xc4d9cbfe60)
        /go/src/github.com/cockroachdb/cockroach/server/updates.go:213 +0xce
github.com/cockroachdb/cockroach/server.(*Server).maybeReportUsage(0xc420307b80, 0x3518a7a786e, 0x22e5940)
        /go/src/github.com/cockroachdb/cockroach/server/updates.go:238 +0x47
github.com/cockroachdb/cockroach/server.(*Server).PeriodicallyCheckForUpdates.func1()
        /go/src/github.com/cockroachdb/cockroach/server/updates.go:97 +0xa8
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc42035f050, 0xc420a35fd0)
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:187 +0x7d
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:188 +0x66
```

--- a/server/updates.go
+++ b/server/updates.go
@@ -208,11 +208,11 @@ func (s *Server) checkForUpdates() {
 
 func (s *Server) usageReportingEnabled() bool {
 	// Grab the optin value from the database.
-	var ctx context.Context
+	ctx := context.TODO()
 	req := &serverpb.GetUIDataRequest{Keys: []string{optinKey}}
 	resp, err := s.admin.GetUIData(ctx, req)
 	if err != nil {
-		log.Warning(context.TODO(), err)
+		log.Warning(ctx, err)
 		return false
 	}
 
@@ -223,7 +223,7 @@ func (s *Server) usageReportingEnabled() bool {
 	}
 	optin, err := strconv.ParseBool(string(val.Value))
 	if err != nil {
-		log.Warningf(context.TODO(), "could not parse optin value (%q): %v", val.Value, err)
+		log.Warningf(ctx, "could not parse optin value (%q): %v", val.Value, err)
 		return false
 	}
 	return optin

